src/main.rs

```diff
@@ -486,17 +486,173 @@ fn shell_exec(vm: &VirtualMachine, source: &str, scope: Scope) -> ShellExecResul
     }
 }
 
+struct ShellHelper<'a> {
+    vm: &'a VirtualMachine,
+    scope: Scope,
+}
+
+impl ShellHelper<'_> {
+    fn complete_opt(&self, line: &str) -> Option<(usize, Vec<String>)> {
+        let mut words = vec![String::new()];
+        fn revlastword(words: &mut Vec<String>) {
+            let word = words.last_mut().unwrap();
+            let revword = word.chars().rev().collect();
+            *word = revword;
+        }
+        let mut startpos = 0;
+        for (i, c) in line.chars().rev().enumerate() {
```

Comment:
This for loop is maybe a good candidate to put into a function called `split_line_on_dots` or something alike.